creature/creature.h

```diff
@@ -15,24 +15,28 @@ enum class CreatureType {
     WOLF
 };
 
-class Creature
-{
+class CreatureManager;
```

Comment:
Didn't get why you've declared the class twise

Reply:
Classes forward declaration